CRM/Admin/Form/Setting/Debugging.php

```diff
@@ -54,4 +55,17 @@ public function buildQuickForm() {
     parent::buildQuickForm();
   }
 
+  /**
+   * Process the form submission.
+   */
+  public function postProcess() {
+    $params = $this->controller->exportValues($this->_name);
+
+    if ($params['isProductionEnvironment'] === FALSE) {
```

Comment:
@eileenmcnaughton I'm not quite sure what you are getting at here: isn't Civi::settings()->set() merely a wrapper for functionality also exposed via API wrapper? Or is your concern that the success message should be bubbled up from something the API calls too?